tracer/src/Datadog.Trace/Configuration/ImmutableTracerSettings.cs

```diff
@@ -587,6 +589,26 @@ public static ImmutableTracerSettings FromDefaultSources()
             return new ImmutableTracerSettings(TracerSettings.FromDefaultSourcesInternal(), true);
         }
 
+        // Overriding the default "record" behaviour here
+        // This type _shouldn't_ be treated as a record generally, we only made it a record
+        // so we could use with {} expressions...
```

Comment:
Nitpicky, but maybe say that without overriding them they were accessing the public properties just to be clear of the why this was done

Reply:
+1